pyriemann/transferlearning.py

```diff
@@ -0,0 +1,158 @@
+        each of the class centroid in parallel.
+        If -1 all CPUs are used. If 1 is given, no parallel computing code is
+        used at all, which is useful for debugging. For n_jobs below -1,
+        (n_cpus + 1 + n_jobs) are used. Thus for n_jobs = -2, all CPUs but one
+        are used.
+
+    Attributes
+    ----------
+    covmeans_ : list
+        Class centroids, estimated after fit.
+    classes_ : list
+        List of classes, obtained after fit
+
+    See Also
+    --------
+    MDM
+
+    References
+    ----------
+    .. [1] E. Kalunga, S. Chevallier and Q. Barthelemy, "Transfer learning for
+        SSVEP-based BCI using Riemannian similarities between users", in 26th
+        European Signal Processing Conference (EUSIPCO), pp. 1685-1689. IEEE,
+        2018.
+
+    .. [2] S. Khazem, S. Chevallier, Q. Barthelemy, K. Haroun and C. Nous,
+        "Minimizing Subject-dependent Calibration for BCI with Riemannian
+        Transfer Learning", in 10th International IEEE/EMBS Conference on
+        Neural Engineering (NER), pp. 523-526. IEEE, 2021.
+    """
+
+    def __init__(self, transfer_coef, metric='riemann', n_jobs=1):
+        """Init."""
+        self.transfer_coef = transfer_coef
+        self.metric = metric
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y, X_source, y_source, sample_weight=None):
+        """Fit (estimates) the centroids.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices from target domain.
+        y : ndarray, shape (n_matrices,)
+            Labels for each matrix of target domain.
+        X_source : ndarray, shape (n_matrices_source, n_channels, n_channels)
+            Set of SPD matrices from source domain.
+        y_source : ndarray, shape (n_matrices_source,)
+            Labels for each matrix of source domain.
+        sample_weight : None | ndarray, shape (n_matrices_source,), default=None
+            Weights for each matrix from the source domain.
+            If None, it uses equal weights.
+
+        Returns
+        -------
+        self : MDWM instance
+            The MDWM instance.
+        """
+
+        if not 0 <= self.transfer_coef <= 1:
+            raise ValueError(
+                'Value transfer_coef must be included in [0, 1] (Got %d)'
+                % self.transfer_coef)
+
+        if isinstance(self.metric, str):
+            self.metric_mean = self.metric
+            self.metric_dist = self.metric
+        elif isinstance(self.metric, dict):
+            # check keys
+            for key in ['mean', 'distance']:
+                if key not in self.metric.keys():
+                    raise KeyError('metric must contain "mean" and "distance"')
+
+            self.metric_mean = self.metric['mean']
+            self.metric_dist = self.metric['distance']
+        else:
+            raise TypeError('metric must be dict or str')
```

Comment:
You should rebase your branch on the last master and use function `_check_metric` added in #181.
```suggestion
        self.metric_mean, self.metric_dist = _check_metric(self.metric)
```